[CI] MixedClusterClientYamlTestSuiteIT test {p0=mtermvectors/10_basic/Tests catching other exceptions per item} failing

**Build Scans:**
- [elasticsearch-intake #42927 / 9.4.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2wklxntc4body)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v9.4.4#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}" -Dtests.seed=E6978A35B462B52 -Dtests.bwc=true -Dtests.locale=hy-AM -Dtests.timezone=Atlantic/Canary -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dmtermvectors%2F10_basic%2FTests%20catching%20other%20exceptions%20per%20item%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: org.junit.internal.runners.model.MultipleFailureException: There were 2 errors:
  java.lang.AssertionError(field [docs.0.term_vectors.text.terms.brown.term_freq] is null)
  java.lang.AssertionError(field [docs.0.term_vectors.text.terms.brown.ttf] is null)
```

**Issue Reasons:**
- [main] 1 consecutive failures in step 9.4.4_bwc-snapshots
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -725,9 +725,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT
   method: testMultiTermVectorsApiRealtimeGet
   issue: https://github.com/elastic/elasticsearch/issues/150101
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
-  issue: https://github.com/elastic/elasticsearch/issues/153084
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:inlinestats.doubleShadowingWithEval}
   issue: https://github.com/elastic/elasticsearch/issues/153096

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mtermvectors/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mtermvectors/10_basic.yml
@@ -101,6 +101,14 @@ setup:
         index: testidx2
         name: test_alias
 
+  # Creating testidx2 can trigger shard rebalancing in mixed-cluster BWC tests.
+  # Wait for relocations to finish so the mtermvectors request hits a fully
+  # recovered shard copy of testidx.
+  - do:
+      cluster.health:
+        index: testidx
+        wait_for_no_relocating_shards: true
+
   - do:
         mtermvectors:
           "term_statistics" : true

